### PR TITLE
Pin max dask and distributed versions to 2021.09.1

### PR DESCRIPTION
**Pull request by galipremsagar**
Changes to be in-line with: https://github.com/rapidsai/cudf/pull/9286/

### CODE REVIEW

**Review by ajschmidt8 — APPROVED** [review]

File: conda/environments/cugraph_dev_cuda11.0.yml [hunk 1]
```diff
@@ -10,8 +10,8 @@ dependencies:
 - libcudf=21.10.*
 - rmm=21.10.*
 - librmm=21.10.*
-- dask>=2021.6.0
-- distributed>=2021.6.0
+- dask>=2021.6.0,<=2021.09.1
+- distributed>=2021.6.0,<=2021.09.1
```
Comment (galipremsagar) [inline conda/environments/cugraph_dev_cuda11.0.yml:14]:
```suggestion
- dask=2021.09.1
- distributed=2021.09.1
```

File: conda/environments/cugraph_dev_cuda11.2.yml [hunk 2]
```diff
@@ -10,8 +10,8 @@ dependencies:
 - libcudf=21.10.*
 - rmm=21.10.*
 - librmm=21.10.*
-- dask>=2021.6.0
-- distributed>=2021.6.0
+- dask>=2021.6.0,<=2021.09.1
+- distributed>=2021.6.0,<=2021.09.1
```
Comment (galipremsagar) [inline conda/environments/cugraph_dev_cuda11.2.yml:14]:
```suggestion
- dask=2021.09.1
- distributed=2021.09.1
```

File: conda/environments/cugraph_dev_cuda11.4.yml [hunk 3]
```diff
@@ -10,8 +10,8 @@ dependencies:
 - libcudf=21.10.*
 - rmm=21.10.*
 - librmm=21.10.*
-- dask>=2021.6.0
-- distributed>=2021.6.0
+- dask>=2021.6.0,<=2021.09.1
+- distributed>=2021.6.0,<=2021.09.1
```
Comment (galipremsagar) [inline conda/environments/cugraph_dev_cuda11.4.yml:14]:
```suggestion
- dask=2021.09.1
- distributed=2021.09.1
```

File: conda/recipes/cugraph/meta.yaml [hunk 4]
```diff
@@ -37,8 +37,8 @@ requirements:
     - cudf={{ minor_version }}
     - dask-cudf {{ minor_version }}
     - dask-cuda {{ minor_version }}
-    - dask>=2021.6.0
-    - distributed>=2021.6.0
+    - dask>=2021.6.0,<=2021.09.1
+    - distributed>=2021.6.0,<=2021.09.1
```
Comment (galipremsagar) [inline conda/recipes/cugraph/meta.yaml:41]:
```suggestion
    - dask=2021.09.1
    - distributed=2021.09.1
```

**Comment by galipremsagar** [comment]
@gpucibot merge

**Comment by galipremsagar** [comment]
@gpucibot merge

**Comment by ajschmidt8** [comment]
@gpucibot merge

**Comment by ajschmidt8** [comment]
@gpucibot merge

**Pull request merged**